lib/cartopy/mpl/feature_artist.py

```diff
@@ -198,20 +200,9 @@ def draw(self, renderer, *args, **kwargs):
 
             stylised_paths.setdefault(style, []).extend(geom_paths)
 
-        transform = ax.projection._as_mpl_transform(ax)
+        self.set_clip_path(ax.patch)
 
-        # Draw one PathCollection per style. We could instead pass an array
-        # of style items through to a single PathCollection, but that
-        # complexity does not yet justify the effort.
         for style, paths in stylised_paths.items():
-            style = style_finalize(dict(style))
-            # Build path collection and draw it.
-            c = matplotlib.collections.PathCollection(paths,
-                                                      transform=transform,
-                                                      **style)
-            c.set_clip_path(ax.patch)
-            c.set_figure(ax.figure)
-            c.draw(renderer)
-
-        # n.b. matplotlib.collection.Collection.draw returns None
-        return None
+            style = dict(style)
+            with self._cm_set(paths=paths, **style):
```

Comment:
I guess I shouldn't really be using `_cm_set` here as it's private to Matplotlib.  It shouldn't be too much work to write a few lines that do the same thing though.